_plugins/bidirectional_links_generator.rb

```diff
@@ -39,108 +39,50 @@ def generate(site)
         e.content.include?(current_note.url)
       end
 
-      # Nodes: Graph
-      graph_nodes << {
-        id: note_id_from_note(current_note),
-        path: current_note.url,
-        label: current_note.data['title'],
-      } unless current_note.path.include?('_notes/index.html')
-
-			# Edges: Jekyll
-      current_note.data['backlinks'] = notes_linking_to_current_note
-
-      # Edges: Graph
-      notes_linking_to_current_note.each do |n|
-        graph_edges << {
-          source: note_id_from_note(n),
-          target: note_id_from_note(current_note),
-        }
-
-        if current_note.data['tags'].include?("CrossFit")
-          crossfit_graph_nodes << {
+      sub_graph_nodes.each {|key, value| 
+        if current_note.data['tags'].include?("#{key}")
+          value << {
             id: note_id_from_note(current_note),
             path: current_note.url,
             label: current_note.data['title'],
           } unless current_note.path.include?('_notes/index.html')
         end
+        value.uniq!
+      }
```

Comment:
iterating through each sub_graph and checking to see if the current note has a tag that matches

Reply:
if it does, it adds the note to the subgraph array